core/src/main/java/com/novoda/merlin/receiver/ConnectivityCallbacks.java

```diff
@@ -35,15 +36,9 @@ public void onLost(Network network) {
     }
 
     private void notifyMerlinService(Network network) {
-        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(network);
-        if (null != networkInfo) {
-            boolean connected = networkInfo.isConnected();
-            String reason = networkInfo.getReason();
-            String extraInfo = networkInfo.getExtraInfo();
-
-            connectivityChangesListener.onConnectivityChanged(ConnectivityChangeEvent.createWithNetworkInfoChangeEvent(connected, extraInfo, reason));
-        } else {
-            connectivityChangesListener.onConnectivityChanged(ConnectivityChangeEvent.createWithoutConnection());
+        if (connectivityChangesNotifier.canNotify()) {
```

Comment:
Both `ConnectivityCallbacks` and `ConnectivityReceiverConnectivityChangeNotifier` check that they are able to `notify`, where `notify` is where the `Service` is bound. If the `Service` is bound then it's `Forwarder` must be present